src/app/static/src/app/components/modelOutput/ModelOutput.vue

```diff
@@ -223,6 +225,17 @@
                     disaggLabel: i18next.t("disaggBy", this.currentLanguage),
                     filters: this.barchartFilters
                 }
+            },
+            flattenedFilterOptionIds(){
+                const ids: Dict<string[]> = {}
+                this.barchartFilters.forEach(filter => {
+                    if ((filter.options[0] as NestedFilterOption).children){
+                        ids[filter.id] = Object.keys(flattenOptions(filter.options))
+                    } else {
+                        ids[filter.id] = filter.options.map((option: FilterOption) => option.id)
+                    }
+                })
```

Comment:
Sorry, I really meant that this just needed to be for the nested filter options, not all filters. But, actually, thinking about it -if it's only ever required for the xAxis filter, it could just be `flattenedXAxisFilterOptionIds` and be recalculated only when x axis filter changes, whether that's nested or not. Might need to amend that later if we end up doing something similar with the disaggregateBy filter to make it consistent.